src/compiler/compile_engine.cc

```diff
@@ -202,6 +202,24 @@ class CompileEngine {
       Array<Tensor> out = fcompute[inode.source->op()](
           inode.source->attrs, op_inputs, out_info);
       CHECK_EQ(out.size(), inode.source->num_outputs());
+
+      // check output dimentions also match
+      // This check is to make sure the NNVM operator Infer match with Compute result.
+      // Missing this check may pass the build but leads to runtime errors.
+      for (uint32_t i=0; i < out.size(); ++i) {
+        CHECK_EQ(out[i].ndim(), out_info[i].ndim()) << inode.source->op()->name;
+        tvm::Tensor inferred_tensor = out[i];
+        tvm::Tensor computed_tensor = out_info[i];
+        // TODO(srkreddy1238): To be verified before enabling it.
```

Comment:
Remove the following lines, output shape cannot equal input shape

Reply:
I don't understand here. No need to check shapes ??

Reply:
I mean the comment, no need to check data types equal

Reply:
I am just confusing about the TODO left here.

Reply:
oh, ok

Reply:
Done